api/base_controller.go

```diff
@@ -282,27 +297,34 @@ func (c *BaseController) DeleteSingleObject(r *web.Request) (*web.Response, erro
 		ResourceType:  c.objectType,
 		PlatformID:    types.SMPlatform,
 		CorrelationID: log.CorrelationIDFromContext(ctx),
-		Context:       &types.OperationContext{Async: c.shouldExecuteAsync(r)},
+		Context:       c.getOperationContextByRequest(r),
 	}
 
-	if c.shouldExecuteAsync(r) {
-		log.C(ctx).Debugf("Request will be executed asynchronously")
+	if operation.IsAsyncResponse() {
+		log.C(ctx).Debugf("Request will be executed asynchronously due to client request async=true")
 		if err := c.checkAsyncSupport(); err != nil {
 			return nil, err
 		}
-
-		if err := c.scheduler.ScheduleAsyncStorageAction(ctx, operation, action); err != nil {
-			return nil, err
-		}
-
-		return util.NewLocationResponse(operation.GetID(), objectID, c.resourceBaseURL)
+		return c.executeAsync(ctx, operation, action, objectID)
 	}
 
-	log.C(ctx).Debugf("Request will be executed synchronously")
 	if _, err := c.scheduler.ScheduleSyncStorageAction(ctx, operation, action); err != nil {
 		return nil, util.HandleStorageError(err, c.objectType.String())
 	}
 
+	operation, err = getLastOperation(ctx, objectID, c.repository)
```

Comment:
why you need to fetch the last operation?
is not the operation that you passed to ScheduleSyncStorageAction function?